packages/forms/src/UIForm/fields/File/File.component.js

```diff
@@ -0,0 +1,125 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { translate } from 'react-i18next';
+import FieldTemplate from '../FieldTemplate';
+import { generateDescriptionId, generateErrorId } from '../../Message/generateId';
+
+import { I18N_DOMAIN_FORMS } from '../../../constants';
+import theme from './File.scss';
+
+class FileWidget extends React.Component {
+	constructor(props) {
+		super(props);
+		// Extract file name from value
+		if (props.value && props.value.indexOf(';name=') !== -1) {
+			const fileName = props.value.slice(props.value.indexOf(';name=') + 6, props.value.indexOf(';base64,'));
+			this.state = { fileReplaceValue: fileName };
```

Comment:
please extract that outside of the component.
What if the props.value change during the life of the FileWidget ?

Reply:
While component mount, the file name is extracted from the data url (`data:text/xml;name=test.xml;base64,PD94bWwgdmVyc2l...`) and inserted in the form.

While `props.value` change during the life of the widget, in `onChange`, the new file name is taken from the file object and added in the form.